### PR TITLE
[ci/docker/air] Update ML/DL dependencies to latest releases / Remove Py3.6 Docker images

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

This PR upgrades some of our dependencies to the latest releases.

At the moment, we are testing with a lower bound of dependencies. This is good as we ensure compatibility with these versions, but at the same time we don't test compatibility with more recent releases. We prioritize backwards compatibility over compatibility with more recent versions.

This PR:
- Introduces a set of _legacy dependencies_ (the current versions in the CI).
- This is a lower bound of dependencies that we guarantee compatibility with
- The regular dependencies are upgraded to more recent releases
- We remove Docker builds for Python 3.6. as this is incompatible with more recent versions of tensorflow/torch
- Wheel building for 3.6 is not affected.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by krfricke** [comment]
Tests are passing except for some in RLlib - I will work with the rllib team to resolve this.

But the rest is ready for review

cc @pcmoritz @richardliaw  for general testing infrastructure
cc @gjoliver @sven1977 for rllib compatibility
cc @amogkam for air compatibility and dependencies

The base CI GPU image has been updated to support CUDA 11.6 already.

File: docker/ray-ml/Dockerfile [hunk 1]
```diff
@@ -23,7 +23,7 @@ RUN sudo apt-get update \
         unzip \
         unrar \
     && $HOME/anaconda3/bin/pip --no-cache-dir install -U pip \
-    && $HOME/anaconda3/bin/pip --no-cache-dir install -U \
+    && $HOME/anaconda3/bin/pip --no-cache-dir install --use-deprecated=legacy-resolver -U \
```
Comment (amogkam) [inline docker/ray-ml/Dockerfile:26]: can we not use the legacy resolver? what are the conflicts we are getting if we remove this?

**Review by amogkam — CHANGES_REQUESTED** [review]

File: python/requirements/ml/requirements_tune.txt [hunk 2]
```diff
@@ -11,8 +11,6 @@ freezegun==1.1.0
 gluoncv==0.10.1.post0
 gpy==1.10.0
 autorom[accept-rom-license]
-gym>=0.21.0,<0.24.0; python_version >= '3.7'
```
[the remark this answers is not in the document]
Comment (sven1977) [reply python/requirements/ml/requirements_tune.txt:14]: Oh, got it. Already in RLlib.

**Review by sven1977 — APPROVED** [review]
LGTM.

**Comment by krfricke** [comment]
Deferring this until https://github.com/ray-project/ray/pull/29375 is merged

**Comment by ddelange** [comment]
I guess this PR could incorporate, supercede and close #26072?

**Comment by ddelange** [comment]
totally fine by me :)

**Review by amogkam — APPROVED** [review]
Thanks @krfricke, lgtm!

**Review by richardliaw — APPROVED** [review]

File: rllib/algorithms/a3c/tests/test_a3c.py [hunk 3]
```diff
@@ -27,7 +27,7 @@ def test_a3c_compilation(self):
         num_iterations = 2
 
         # Test against all frameworks.
-        for _ in framework_iterator(config, with_eager_tracing=True):
+        for _ in framework_iterator(config, with_eager_tracing=False):
```
Comment (ArturNiederfahrenhorst) [inline rllib/algorithms/a3c/tests/test_a3c.py:30]: @gjoliver @kouroshHakha Not sure why this test did not fail initially, but A3C seems error with eager tracing as well, right @krfricke ?

**Pull request merged**